### PR TITLE
Add Microsoft.NET.HostModel.TestData

### PR DESCRIPTION
**Pull request by jtschuster**
Adds a project with binary Mach-O files to test the ad-hoc signing in Microsoft.NET.HostModel (https://github.com/dotnet/runtime/pull/108992).

### CODE REVIEW

File: src/Microsoft.NET.HostModel.TestData/Microsoft.NET.HostModel.TestData.sln [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿
```
Comment (akoeplinger) [inline src/Microsoft.NET.HostModel.TestData/Microsoft.NET.HostModel.TestData.sln:1]: we usually don't have .sln files committed in this repo

File: src/Microsoft.NET.HostModel.TestData/data/src/main.c [hunk 2]
```diff
@@ -0,0 +1,9 @@
+int add(int a, int b)
```
Comment (akoeplinger) [inline src/Microsoft.NET.HostModel.TestData/data/src/main.c:1]:
```suggestion
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

int add(int a, int b)
```

File: src/Microsoft.NET.HostModel.TestData/Microsoft.NET.HostModel.TestData.csproj [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+    <LangVersion>latest</LangVersion>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <EmbeddedResource Include="data\*" />
```
[the remark this answers is not in the document]
Comment (akoeplinger) [reply src/Microsoft.NET.HostModel.TestData/Microsoft.NET.HostModel.TestData.csproj:11]: I agree

[the remark this answers is not in the document]
Comment (jtschuster) [reply src/Microsoft.NET.HostModel.TestData/Microsoft.NET.HostModel.TestData.csproj]: Changed this to follow the format of the other projects in the repo.

**Review by akoeplinger — APPROVED** [review]

File: src/Microsoft.NET.HostModel.TestData/Microsoft.NET.HostModel.TestData.csproj [hunk 4]
```diff
@@ -0,0 +1 @@
+﻿<Project Sdk="Microsoft.Build.NoTargets" />
```
[the remark this answers is not in the document]
Comment (jtschuster) [reply src/Microsoft.NET.HostModel.TestData/Microsoft.NET.HostModel.TestData.csproj:1]: I'll remove them just to be safe.

**Pull request merged**